src/azdata.proposed.d.ts

```diff
@@ -397,9 +386,23 @@ declare module 'azdata' {
 	}
 
 	export interface ConnectionProvider extends DataProvider {
+		/**
+		 * Changes the password for a connection profile.
+		 */
 		changePassword(connectionUri: string, connectionInfo: ConnectionInfo, newPassword: string): Thenable<PasswordChangeResult>;
 	}
 
+	export interface PasswordChangeResult {
+		/**
+		 * Whether the password change was successful
+		 */
+		result: boolean;
+		/**
+		 * Error message if the password change was unsuccessful
+		 */
+		errorMessage?: string | undefined;
```

Comment:
having the `?` mark is the indicator nullable, so you don't have to add the `| undefined` type